app/models/date_of_birth.rb

```diff
@@ -0,0 +1,71 @@
+class DateOfBirth < Base
+
+  attribute :date_of_birth, :datetime
+  
+  validates :date_of_birth, presence: { message: "not a valid date" }
+  
+  def next_step
+   "uk_or_overseas" 
+  end
+
+  def initialize(attributes = nil)
+    super
+
+    assign_attributes(attributes) if attributes
+  end
+
+  def assign_attributes(attributes)
+    new_attributes = attributes.stringify_keys
+    multiparameter_attributes = extract_multiparameter_attributes(new_attributes)
+
+    multiparameter_attributes.each do |multiparameter_attribute, values_hash|
+      set_values = (1..3).collect{ |position| values_hash[position].to_i }
+
+    # validation checks
+    if set_values.include?(0)
+      return
+    elsif set_values[0] < 1920 || set_values[0] > Date.today.year
+      return
+    elsif set_values[1] < 1 || set_values[1] > 12
+      return
+    elsif set_values[2] < 1 || set_values[2] > 31
+      return
+    end
+  
+      self.send("#{multiparameter_attribute}=", Date.new(*set_values))
+    end
+  end
+
+  protected
+
+  def extract_multiparameter_attributes(new_attributes)
```

Comment:
I'm probably missing something here, but does this need to be so complicated? It looks like the input is a hash:

```
{'date_of_birth(3i)' => '1', 'date_of_birth(2i)' => '12', 'date_of_birth(1i)' => '2001'}
```
Why can't we just do:

```
Date.new(hash['date_of_birth(1i)'], hash['date_of_birth(12)'], hash['date_of_birth(3i)'])
```

Reply:
I'm probably missing something myself, I've played around with this for a while, the simple solution would be a custom form with day, month and year params but the gov builder form sends out multiparameter attributes, and this seems to be the standard way of dealing with them in ActiveModel.

Reply:
Ok, I don't want to hold up your PR on it - if I get some time I'll take a proper look at wrap my head around it